components/layout/positioned.rs

```diff
@@ -248,13 +202,57 @@ impl PositioningContext {
         new_fragment
     }
 
+    fn take_boxes_for_fragment(
+        &mut self,
+        new_fragment: &BoxFragment,
+        boxes_to_layout_out: &mut Vec<HoistedAbsolutelyPositionedBox>,
+        boxes_to_continue_hoisting_out: &mut Vec<HoistedAbsolutelyPositionedBox>,
+    ) {
+        debug_assert!(
+            new_fragment
+                .style
+                .establishes_containing_block_for_absolute_descendants(new_fragment.base.flags)
+        );
+
+        if new_fragment
+            .style
+            .establishes_containing_block_for_all_descendants(new_fragment.base.flags)
+        {
+            boxes_to_layout_out.append(&mut self.absolutes);
+            return;
+        }
+
+        // TODO: This could potentially use `extract_if` when that is stabilized.
```

Comment:
(this is happening in Rust 1.87 in a couple of weeks)

Reply:
I saw that! There's a few thing we can clean up when it's finally no longer nightly-only.